flight/Modules/Logging/logging.c

```diff
@@ -285,6 +352,81 @@ static void loggingTask(void *parameters)
 	}
 }
 
+
+/**
+ * Log all settings objects
+ * \param[in] obj Object to log
+ */
+static void logSettings(UAVObjHandle obj)
+{
+	if (UAVObjIsSettings(obj)) {
+		UAVTalkSendObjectTimestamped(uavTalkCon, obj, 0, false, 0);
+	}
+}
+
+/**
+ * Write log file header
+ * see firmwareinfotemplate.c
+ */
+static void writeHeader()
+{
+	int pos;
+	char tmp_str[45];
+	char *info_str;
+	char this_char;
+	DateTimeT date_time;
+
+	const struct pios_board_info * bdinfo = &pios_board_info_blob;
+
+	sprintf(tmp_str, "%s\n", "Tau Labs git hash:");
+	send_data((uint8_t*)tmp_str, strlen(tmp_str));
+
+	// Commit tag name
+	info_str = (char*)(bdinfo->fw_base + bdinfo->fw_size + 14);
+	send_data((uint8_t*)info_str, strlen(info_str));
+
+	// Git commit hash
+	pos = 0;
+	tmp_str[pos++] = ':';
+	for (int i = 0; i < 4; i++){
+		this_char = *(char*)(bdinfo->fw_base + bdinfo->fw_size + 7 - i);
+		tmp_str[pos++] = DIGITS[(this_char & 0xF0) >> 4];
+		tmp_str[pos++] = DIGITS[(this_char & 0x0F)];
+	}
+	send_data((uint8_t*)tmp_str, pos);
+
+	// Date
+	date_from_timestamp(*(uint32_t *)(bdinfo->fw_base + bdinfo->fw_size + 8), &date_time);
+	sprintf(tmp_str, " %d%02d%02d\n", 1900 + date_time.year, date_time.mon + 1, date_time.mday);
```

Comment:
snprintf is nice, both because it precludes any chance of an overflow and because it returns the length, avoiding a strlen.